config/config.go

```diff
@@ -884,6 +906,57 @@ func (p *PackageConfig) Validate(path string) error {
 	return p.cachedErr
 }
 
+// SanitizeName forms the package name for the symlink/filepath of the package on the system.
+func (p *PackageConfig) SanitizeName() string {
+	return fmt.Sprintf("%s-%s", strings.ReplaceAll(p.Package, "/", "-"), p.SanitizeVersion())
+}
+
+// SanitizeVersion replaces the version string.
+func (p *PackageConfig) SanitizeVersion() string {
+	// replaces all the . if they exist with _
+	return strings.ReplaceAll(p.Version, ".", "_")
+}
+
+// GenerateFilePath generates what the expected filepath of the package is on the system.
+// based on the package name.
+func (p *PackageConfig) GenerateFilePath() string {
+	// first get the base root
+
+	// for backwards compatibility, packages right now don't have ml_models as a type.
+	// so if that is the case we need to join it right now to packages/../name
+	var dir string
+	if p.Type == "" {
+		// then this is not set and it must be an ml-model for backward compatibility -- but for now just join it without the type
+		// package manager will still create symlinks for these packages based on the packges path
+		dir = path.Clean(path.Join(viamDotDir, "packages", p.Name))
+	} else {
+		dir = path.Clean(path.Join(viamDotDir, "packages", p.GetPackageDirectoryFromType(), dataDotDir, p.SanitizeName()))
```

Comment:
viamDotDir and dataDotDir are variables, so I think "packages" should be one as well for consistency.

Reply:
added!